core/src/main/java/org/elasticsearch/index/query/SpanMultiTermQueryBuilder.java

```diff
@@ -62,11 +63,21 @@ protected void doXContent(XContentBuilder builder, Params params)
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
         Query subQuery = multiTermQueryBuilder.toQuery(context);
+        float boost = 1f;
+        if (subQuery instanceof BoostQuery) {
+            BoostQuery boostQuery = (BoostQuery) subQuery;
+            subQuery = boostQuery.getQuery();
+            boost = boostQuery.getBoost();
+        }
         if (subQuery instanceof MultiTermQuery == false) {
             throw new UnsupportedOperationException("unsupported inner query, should be " + MultiTermQuery.class.getName() +" but was "
                     + subQuery.getClass().getName());
         }
-        return new SpanMultiTermQueryWrapper<>((MultiTermQuery) subQuery);
+        Query wrapper = new SpanMultiTermQueryWrapper<>((MultiTermQuery) subQuery);
+        if (boost != 1f) {
+            wrapper = new BoostQuery(wrapper, boost);
```

Comment:
+1 to SpanBoostQuery. I suspect it never fails because we have no test that uses a span-multi query as an inner query (as opposed to top-level)?